### PR TITLE
Added option to display both absolute and percentage values in pie chart.

### LINKED ISSUE #84: Add an option to display both absolute and percentage values in pie chart
When you want to draw a pie chart, you have the option to display absolute or percentage values on the chart.
- For displaying a value as absolute you set `$Format['WriteValues']` to `PIE_VALUE_NATURAL`;
- For displaying a value as percentage you set `$Format['WriteValues']` to `PIE_VALUE_PERCENTAGE`;

I'd like to add an option to display both absolute and percentage values on the chart. I propose to display this as follows:
```
4
(20%)
```
I propose to set `$Format['WriteValues']` to `PIE_VALUE_BOTH` when you want both absolute and percentage values to be displayed.

I will create a pull request shortly.

### PR DESCRIPTION
**Pull request by MegaChriz**
Fixes #84.

### CODE REVIEW

**Comment by MegaChriz** [comment]
I see that the build is failing but I believe that the failure is a false positive.

```
Line   src/Chart/Pie.php                           

 ------ -------------------------------------------- 

  2353   Variable $Absolute might not be defined.    

  2356   Variable $Percentage might not be defined.  

  2359   Variable $Absolute might not be defined.    

  2359   Variable $Percentage might not be defined.
```
If you inspect the code clearly, then you can see that `$Absolute` and `$Percentage` always will have a value when they are used.
- `$Absolute` is defined when `$WriteValues` is `PIE_VALUE_NATURAL` or `PIE_VALUE_BOTH`. The variable isn't used later on if `$WriteValues` is something else.
- `$Percentage` is defined when `$WriteValues` is `PIE_VALUE_PERCENTAGE` or `PIE_VALUE_BOTH`. The variable isn't used later on if `$WriteValues` is something else.

But perhaps I could write the code differently, if desired. I think I would fix it by defining two additional methods.

**Comment by szymach** [comment]
Hello!
Thank you for your submission. I will have to consider this addition a bit, I think I will be able to do it this weekend. Cheers.

File: src/Chart/Pie.php [hunk 1]
```diff
@@ -2338,4 +2318,48 @@ public function clean0Values(array $Data, array $Palette, $DataSerie, $AbscissaS
...
+     *   The number of decimal digits to round to, in case of displaying a percentage.
+     * @param string $ValueSuffix
+     *   The text to display after the value, in case an absolute value gets displayed.
+     *
+     * @return string
+     *   The value to use for display.
+     */
+    protected function getDisplayValue($Value, $WriteValues, $SerieSum, $Precision, $ValueSuffix)
+    {
+        if ($WriteValues == PIE_VALUE_NATURAL || $WriteValues == PIE_VALUE_BOTH) {
```
Comment (szymach) [inline src/Chart/Pie.php:2344]: Hello, sorry for late reply. Try the code below, it should silence any static code exceptions.

```php
$ValidWriteValues = [PIE_VALUE_NATURAL, PIE_VALUE_PERCENTAGE, PIE_VALUE_BOTH];
if (false === in_array($WriteValues, $ValidWriteValues, true)) {
    return '';
}

$isBothOrNatural = in_array($WriteValues, [PIE_VALUE_NATURAL, PIE_VALUE_BOTH], true);
$isBothOrPercentage = in_array($WriteValues, [PIE_VALUE_PERCENTAGE, PIE_VALUE_BOTH], true);

$Absolute = true === $isBothOrNatural ? ($Value . $ValueSuffix) : null;
$Percentage = true === $isBothOrPercentage
    ? sprintf('%s%', round((100 / $SerieSum) * $Value, $Precision))
    : null
;

if (null === $Absolute && null === $Percentage) {
    throw new RuntimeException(
        "Neither absolute nor percentage display value was calculated"
        . " for display value \"{$WriteValues}\"."
    );
}

if (null !== $Absolute && null !== $Percentage) {
    $Result = "{$Absolute}\n({$Percentage})";
} elseif (null !== $Absolute) {
    $Result = $Absolute;
} elseif (null !== $Percentage) {
    $Result = $Percentage;
}

return $Result;
```
Also, can you please write a test case in [this](https://github.com/szymach/c-pchart/blob/3.0/tests/unit/PieTest.php) unit test for the new option?

**Comment by szymach** [comment]
Hello!
I am closing this PR, since I have incorporated Your code myself into the `3.1` branch. I will make a release sometime soon. 

Cheers.

**Pull request closed without merging**